### PR TITLE
1.4/1699 exp add linting

### PR DESCRIPTION
**Pull request by charlesstrange2**
OK, so this is the result of linting and prettying an entire project at once

https://trello.com/c/nSRDDFvA/1669-add-linting-prettier-on-commit-ee

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +52,21 @@
     "scripts": {
         "dev": "cross-env NODE_ENV=development webpack-dev-server --open --hot",
         "build": "cross-env NODE_ENV=production webpack --mode production --progress --hide-modules --display-error-details",
-        "test": "testcafe chrome,firefox ./tests/*/*.test.js --app 'cross-env NODE_ENV=development webpack-dev-server' --app-init-delay 10000"
+        "test": "testcafe chrome,firefox ./tests/*/*.test.js --app 'cross-env NODE_ENV=development webpack-dev-server' --app-init-delay 10000",
+        "lint": "eslint --ext .js,.vue src tests",
+        "lint-autofix": "eslint --ext .js,.vue src tests --fix",
```
Comment (katjam) [inline package.json:57]: Do you need the `.` to specify all recursively? https://eslint.org/docs/user-guide/command-line-interface#-ext

Also - have found that if you lint on commit, then people can manually lint with eslit directly and don't need to specify all these as scripts.

If you do keep them, they should be in sync with your staged command (see below)

File: package.json [hunk 2]
```diff
@@ -43,7 +52,21 @@
...
-    "version": "1.4.0-dev"
+    "version": "1.4.0-dev",
+    "husky": {
+        "hooks": {
+            "pre-commit": "lint-staged"
+        }
+    },
+    "lint-staged": {
+        "*.{js,json,yaml,json,vue}": [
+            "eslint --ext .js,.vue src tests --fix --quiet --cache"
```
Comment (katjam) [inline package.json:69]: Should this be `npm run lint-autofix` ?

**Review by katjam — COMMENTED** [review]
Much improved. A couple comments.

**Review by katjam — COMMENTED** [review]
I'm sad about the so many unused imports, but good they can be tidied now.

File: package.json (hunk 1 above)
Comment (charlesstrange2) [reply package.json:57]: these lint commands where there to lint and fix manuallywhile i was kicking the tires. remove?
the staged command has quiet and cache only flags to only lint the current changed files.

File: package.json (hunk 2 above)
Comment (charlesstrange2) [reply package.json:69]: needs the quiet and cache flags to only check then changed files and not make a verbose fuss.

**Comment by charlesstrange2** [comment]
I've resolved the outstanding complaints that the vue/essential rules had
- fixed the missing keys
- unwound the impure functions.

**Comment by charlesstrange2** [comment]
closing to recreate new branch

**Pull request closed without merging**